gatling-http/src/main/scala/io/gatling/http/cookie/CookieJar.scala

```diff
@@ -138,4 +138,27 @@ private[http] final case class CookieJar(store: Map[CookieKey, StoredCookie]) {
         }
         .map(_._2.cookie)
     }
+
+  private def isSecureContext(uri: Uri): Boolean =
+    if (uri.getScheme == Uri.HTTPS || uri.getScheme == Uri.WSS) {
+      true
+    } else {
+      val host: String = uri.getHost
+      if (host == null) {
+        false
+      } else if (host == "[::1]" || host == "[0000:0000:0000:0000:0000:0000:0000:0001]") {
+        // The host matches a CIDR notation of ::1/128// The host matches a CIDR notation of ::1/128
+        true
+      } else if (host.matches("127.\\d{1,3}.\\d{1,3}.\\d{1,3}")) {
```

Comment:
This line is not great performance-wise as you're compiling the underlying Pattern every time this method is executed. You might want to improve the implementation in Keycloak too.

Reply:
Thank you - this will be tracked in https://github.com/keycloak/keycloak/issues/32428